util/Pos/Util/Log.hs

```diff
@@ -0,0 +1,278 @@
+-- | Logging implemented with library `katip`
+
+module Pos.Util.Log
+       (
+       -- * Logging
+         Severity(..)
+       , LogContext
+       , LogContextT
+       , LoggingHandler
+       -- * Compatibility
+       , CanLog(..)
+    --    , HasLoggerName(..)
+       , WithLogger
+       , LoggerNameBox(..)
+       -- * Configuration
+       , LoggerConfig(..)
+       , loadLogConfig
+       , parseLoggerConfig
+       , retrieveLogFiles
+       , setLogBasePath
+       -- * Startup
+       , setupLogging
+       -- * Do logging
+       , loggerBracket
+       , usingLoggerName
+       , usingLoggerNames
+       -- * Functions
+       , logDebug
+       , logInfo
+       , logNotice
+       , logWarning
+       , logError
+       , logMessage
+       -- * Naming/Context
+       , LoggerName
+       , askLoggerName
+       , addLoggerName
+       ) where
+
+import           Universum
+
+import           Control.Lens (each)
+import           Control.Monad.Base (MonadBase)
+import           Control.Monad.Morph (MFunctor (..))
+
+import           Pos.Util.Log.Severity (Severity (..))
+import           Pos.Util.LoggerConfig
+
+import qualified Data.Text as T
+import           Data.Text.Lazy.Builder
+
+import           Pos.Util.Log.Internal (LoggingHandler)
+import qualified Pos.Util.Log.Internal as Internal
+import           Pos.Util.Log.Scribes
+
+import qualified Katip as K
+import qualified Katip.Core as KC
+
+
+-- | alias - pretend not to depend on katip
+type LogContext = K.KatipContext
+type LogContextT = K.KatipContextT
+
+type WithLogger m = (CanLog m) --, HasLoggerName m)
+
+type LoggerName = Text
+
+-- -- | compatibility
+class (MonadIO m, LogContext m) => CanLog m where
+    dispatchMessage :: LoggingHandler -> Severity -> Text -> m ()
+    dispatchMessage _ s t = K.logItemM Nothing (Internal.sev2klog s) $ K.logStr t
+
+-- class (MonadIO m, LogContext m) => HasLoggerName m where
+--     askLoggerName' :: m LoggerName
+--     askLoggerName' = askLoggerName
+--     setLoggerName' :: LoggerName -> m a -> m a
+--     setLoggerName' = modifyLoggerName' . const
+--     modifyLoggerName' :: (LoggerName -> LoggerName) -> m a -> m a
+--     modifyLoggerName' f a = addLoggerName (f "cardano-sl")$ a
+-- instance (Monad m, HasLoggerName m) => HasLoggerName (ReaderT a m) where
+-- instance (Monad m, HasLoggerName m) => HasLoggerName (StateT a m) where
+-- instance (Monoid w, Monad m, HasLoggerName m) => HasLoggerName (WriterT w m) where
+-- instance (Monad m, HasLoggerName m) => HasLoggerName (ExceptT e m) where
+--     askLoggerName'    = lift askLoggerName'
```

Comment:
Can we not delete this commented out code?

Reply:
yes, CBR-361 "remove compatibility layer" will remove `CanLog` and `HasLoggerName`
they become obsolete once we have verified `Trace` is working.